### PR TITLE
aptos: update to build with rust 1.71

### PR DESCRIPTION
**Pull request by 0xmigo**
This is to update the Rust version from the formula to match rust version from our aptos CLI codebase - (Aptos recently has updated rust version from 1.70 to 1.71) - https://github.com/aptos-labs/aptos-core/pull/9260

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/aptos.rb
Comment (branchvincent) [inline Formula/aptos.rb:37]: this should use homebrew's `rust` instead

**Review by branchv — CHANGES_REQUESTED** [review]

**Comment by carlocab** [comment]
> * [x]  Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

Note that `aptos` does not actually pass `brew audit --strict`, so this box should probably not have been checked:
```
❯ brew audit --strict aptos
aptos
  * line 25, col 3: Formulae in homebrew/core should use 'depends_on "rust" => build' instead of 'depends_on "rustup-init" => :build'.
  * line 36, col 12: Formula in homebrew/core should not use `rustup-init` at build-time.
Error: 2 problems in 1 formula detected.
```

**Comment by 0xjinn** [comment]
> > * [x]   Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?
> 
> Note that `aptos` does not actually pass `brew audit --strict`, so this box should probably not have been checked:
> 
> ```
> ❯ brew audit --strict aptos
> aptos
>   * line 25, col 3: Formulae in homebrew/core should use 'depends_on "rust" => build' instead of 'depends_on "rustup-init" => :build'.
>   * line 36, col 12: Formula in homebrew/core should not use `rustup-init` at build-time.
> Error: 2 problems in 1 formula detected.
> ```

Ahh thanks for pointing it out. I thought nothing would break since I only updated the rust version. 

I will go ahead and fix the syntax

**Comment by 0xjinn** [comment]
@carlocab Curious, whats your recommandation for replacing `rustup-init` at build time?

File: Formula/aptos.rb
Comment (gregnazario) [reply Formula/aptos.rb:37]: What version is homebrew's Rust on?  Our codebase is constantly upgrading to the latest rust toolchain, so it has to be up to date, or it may fail.

Comment (branchvincent) [reply Formula/aptos.rb:37]: That should be fine. We track the `stable` channel, so currently 1.71: https://github.com/Homebrew/homebrew-core/blob/master/Formula/rust.rb

Comment (0xjinn) [reply Formula/aptos.rb:37]: In the future, if `stable` is pointing 1.72, but our toolchain is still pointing to 1.71. This probably would fail?

Comment (SMillerDev) [reply Formula/aptos.rb:37]: Yes, which would be a reason for brew to check for compatibility with the new version

Comment (0xjinn) [reply Formula/aptos.rb:37]: Gotcha!

Noob question - How do I point this to homebrew rust stable version instead of hardcoded `1.71`? Is there some kind of env var?

Comment (gregnazario) [reply Formula/aptos.rb:37]: So, if I didn't update my rust, but the CLI was updated, I wouldn't get a CLI update until my Rust was also updated?

Comment (branchvincent) [reply Formula/aptos.rb:37]: You can replace `depends_on "rustup-init" => :build` with `depends_on "rustup" => :build`. And then the install block can be:

```ruby
system "cargo", "install", *std_cargo_args(path: "crates/aptos"), "--profile=cli"
```

Overriding `RUSTFLAGS` isn't needed since it's already in your [`.cargo/config.toml`](https://github.com/aptos-labs/aptos-core/blob/aptos-node-v1.5.2/.cargo/config.toml#L16)

Comment (branchvincent) [reply Formula/aptos.rb:37]: >  I wouldn't get a CLI update until my Rust was also updated?

Assuming you installed Homebrew in the default prefix (as most users do), you could update the two independently since you would be installing from our pre-built binaries (and not need to install/upgrade rust)

Comment (0xjinn) [reply Formula/aptos.rb]: Thank you @branchvincent 

The code has been updated with recommended suggestion.

Comment (0xjinn) [reply Formula/aptos.rb]: Unchecked the brew audit box and revert back to use rustup, because using brew rust is causing the build to fail

Comment (0xjinn) [reply Formula/aptos.rb]: Unchecked the brew audit box and revert back to use rustup, because using brew rust is causing the build to fail

@branchvincent May I get another look again? All checks seems to be passing

**Review by branchv — APPROVED** [review]
What you had before was close, it just looks like cargo had some trouble respecting `.cargo/config.toml`'s `build.rustflags` (built successfully on arm though). Anyway, I pushed a change to keep `RUSTFLAGS`.

Thanks for your first contribution and your patience here! 🍻

**Comment by 0xjinn** [comment]
@branchvincent Nice! Thank you so much for the help!

Is looks like merging is still blocked? Wondering if there anything I am missing before we can merge this in?

**Comment by branchvincent** [comment]
We just need one more review, should be merged soon

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
LGTM, thanks!

**Pull request merged**
